abi-bindings/go/packer/packer_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package packer
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"math/big"
+	"math/rand"
+	"reflect"
+	"testing"
+
+	validatorsetsig "github.com/ava-labs/teleporter/abi-bindings/go/governance/ValidatorSetSig"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/teleporter/TeleporterMessenger"
+	teleporterregistry "github.com/ava-labs/teleporter/abi-bindings/go/teleporter/registry/TeleporterRegistry"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/tools/go/packages"
+)
+
+const interfaceName = "ABIPacker"
+
+// Paths below are relative to the current package.
+const packagePath = "../..."
+const currentPackagePath = "."
+
+var fs = token.NewFileSet()
+
+var packerTypes = map[string]ABIPacker{
+	"ValidatorSetSigMessage": &validatorsetsig.ValidatorSetSigMessage{},
+	"TeleporterMessage":      &teleportermessenger.TeleporterMessage{},
+	"ProtocolRegistryEntry":  &teleporterregistry.ProtocolRegistryEntry{},
+}
+
+func findAllImplementers(t *testing.T) []string {
```

Comment:
What's the advantage of dynamically fetching the implementers if we still have to maintain a hardcoded list of them in `packerTypes`? If a type implements `ABIPacker`, then the exhaustiveness check will be run against it, which fails if it's not also named in `packerTypes`.

Reply:
If you forget to add it to the `packerTypes`, which is very likely without this enforcement mechanism, then the test is rather useless. It only checks for types that we remembered to add to the list. 

Originally I didn't want to require the explicit list and only find them programmatically but I didn't find a way to convert from `types` types that the `"golang.org/x/tools/go/packages"` uses and the `reflect` types that the randomizer uses hence I use the names of the types as a "bridge".